### ISSUE TITLE
Tidy / test `Funds` helper functions

### ISSUE THREAD

**Issue by NiloCK** [statechannels/go-nitro#47]
Are these functions referenced outside of this file?

 I think we could probably have the same name for both functions and that would make it clearer:
```go
// Sum returns the Sum of the reciever and input Funds objects
func (h Funds) Sum(a ...Funds) Funds {
	a = append(a, h)
	return Sum(a...)
}

// Sum returns a new Funds object with all of the asset keys from the supplied Funds objects, each having an amount summed across that asset's amount in each input object.
// e.g. {[0x0a,0x01][0x0b,0x01]} + {[0x0a,0x02]} = {[0x0a,0x03][0x0b,0x01]}
func Sum(a ...Funds) Funds {
	sum := Funds{}

	for _, holdings := range a {
		for asset, amount := range holdings {
			sum[asset] = sum[asset].Add(sum[asset], amount)
		}
	}

	return sum
}
```

I put in an example to help explain the function. I think we ought to also have a test for both functions.

_Originally posted by @geoknee in https://github.com/statechannels/go-nitro/pull/42#discussion_r745502100_

**Comment by NiloCK** [comment]
> They were used outside of this file, but a rewrite of some logic in direct-funing.go removed it.
> 
> Possibly a UK/NA distinction, but a.Sum(b) returning the sum of a and b does not read very naturally to me. Sum wants (grammatically) an unordered set as input, and having one element differently positioned as the receiver doesn't jive.
> 
> Updates to the comments are great. 03807ef
> 
> I've done some work on tests (and associated Equal function) but will park it for a separate PR (#47).
> 

(me, from same comment thread)

**Comment by geoknee** [comment]
I don't think it is particularly a UK/NA thing. I agree it doesn't jive well. 

* a + b is "a plus b" or "a add b". You've got a binary operator.

* \sum _i c_i is "the sum of" the set of c_i 

but your `a.Add(b)` method is actually `a.Add(b_i)`  -- so it's not a binary operator. If you changed it 
```diff
- func (h Funds) Add(a ...Funds)
+ func (h Funds) Add(a Funds)
```
then it would make more sense to me why it might have a different name.

I think in any case this is not worth us overly worrying about :-)

**Comment by NiloCK** [comment]
See also (closed) #50

**Comment by NiloCK** [comment]
closed via #64
